### PR TITLE
KAFKA-12841: Remove an additional call of onAcknowledgement

### PR DESCRIPTION
**Pull request by philipnee**
The bug was introduced in https://github.com/apache/kafka/pull/11689/ that an additional onAcknowledgement was made using the InterceptorCallback class.  This is undesirable since onSendError will attempt to call onAcknowledgement once more.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by junrao — COMMENTED** [review]
@philipnee : Thanks for the PR. Looks good to me. Would it be possible to add a unit test for this?

**Comment by philipnee** [comment]
Hey @junrao - Much thanks to look into this, I added some tests to the PR, let me know if you think the testing strategy is acceptable...

File: clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java [hunk 1]
```diff
@@ -1905,15 +1905,21 @@ public void testCallbackHandlesError() throws Exception {
         Map<String, Object> configs = new HashMap<>();
         configs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9000");
         configs.put(ProducerConfig.MAX_BLOCK_MS_CONFIG, "1000");
+        configs.put(ProducerConfig.INTERCEPTOR_CLASSES_CONFIG, MockProducerInterceptor.class.getName());
```
Comment (junrao) [inline clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java:1908]: Perhaps the method name could be changed to testCallbackAndInterceptorHandleError?

**Review by junrao — COMMENTED** [review]
@philipnee : Thanks for the updated PR. Great test! Just one minor comment.

File: clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java (hunk 1 above)
Comment (philipnee) [reply clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java:1908]: Good call.

**Review by junrao — APPROVED** [review]
@philipnee : Thanks for the updated PR. LGTM. Waiting for the tests to pass.

**Comment by junrao** [comment]
retest this please

**Comment by junrao** [comment]
ok to test

**Comment by junrao** [comment]
test this please

**Review by junrao — COMMENTED** [review]
@philipnee :  Did you push the wrong commit?

**Comment by philipnee** [comment]
Hey @junrao - I think I rebased it onto the wrong head.  It seems fixed now.

**Review by junrao — APPROVED** [review]
@philipnee : Thanks for rebasing. LGTM

**Comment by junrao** [comment]
@cadonna : Could we merge this PR to 3.2? It fixes an issue in 3.2.

**Comment by cadonna** [comment]
@junrao @philipnee Thank you for pinging me! 
Since I have already created RC0 which is under voting I actually only accept regressions and exceptionally severe bugs that justify the creation of a new release candidate. This bug seems to have been around for quite some time and it does not seem to be a regression or exceptionally severe. So I would prefer not to cherry pick this PR to 3.2 before 3.2.0 is released. After the release, you should cherry-pick this PR to 3.2 in order to include it in the bugfix release 3.2.1.
If you still think that this PR should be included in 3.2.0 please raise it on the voting thread for RC0 of 3.2.0 on the dev mailing list.

**Comment by junrao** [comment]
@cadonna : Thanks for the response. This bug was actually introduced in 3.2.0. I will ping the mailing list.

**Comment by cadonna** [comment]
@junrao Now I see how this issue was introduced in 3.2.0. The fix for the bug described in KAFKA-12841 introduced it, right? I initially understood that this PR is the fix for the bug described in KAFKA-12841 which dates back to 2.6.

I think that classifies as a regression. I will cherry-pick this PR to 3.2.0 and create a new release candidate.

**Comment by junrao** [comment]
Thanks, @cadonna. I will mark the jira as fixed in 3.2.0 then.

**Comment by cadonna** [comment]
Cherry-picked to 3.2

**Pull request merged**